src/main/java/org/opensearch/knn/index/query/KNNQueryBuilder.java

```diff
@@ -49,6 +53,8 @@ public class KNNQueryBuilder extends AbstractQueryBuilder<KNNQueryBuilder> {
     private final String fieldName;
     private final float[] vector;
     private int k = 0;
+    private QueryBuilder filter;
+    private static final Version MINIMAL_SUPPORTED_VERSION_FOR_LUCENE_HNSW_FILTER = Version.V_3_0_0;
```

Comment:
Shouldnt this be 2.4?

Reply:
It should be eventually, but only after we merge it to 2.x. As this goes to main first at the moment this is merged 2.4 will not have code for filtering. I think the correct way of setting this min supported version:
merge to main - min version = 3.0
backport to 2.x - min version = 2.4 in 2.x
submit one more PR to main - min version = 2.4